iroh-net/src/discovery.rs

```diff
@@ -692,24 +684,17 @@ mod test_dns_pkarr {
     }
 
     async fn ep_with_discovery(
-        relay_map: RelayMap,
-        nameserver: SocketAddr,
-        node_origin: &str,
-        pkarr_relay: &Url,
+        relay_map: &RelayMap,
```

Comment:
why change this to a reference if it then immediately clones it?  by keeping it as owned it's more obvious to the caller what happens to it and they get to do the clone, if needed, explicitly.

Reply:
oh, i guess this is just a test util.  but still, i find that an odd pattern even if it is for tests.  if you really like the convenience of calling it with borrowed data maybe using a generic with `Clone` or `ToOwned` is better.

Reply:
I agree in general but thought for a test only util it's ok and makes writing tests a bit simpler IMO. But no strong opinion, can also change back to take by value.

Reply:
yeah, i approved anyway as i noticed it's a test.  up to you